ml-agents/mlagents/trainers/trainer.py

```diff
@@ -151,19 +165,58 @@ def export_model(self):
         """
         self.policy.export_model()
 
-    def write_summary(self, global_step, lesson_num=0):
+    def write_training_metrics(self):
+        """
+        Write training metrics to a CSV  file
+        :return:
+        """
+        log_file = self.summary_path + '.csv'
+        with open(log_file, 'w') as f:
+            f.writelines('%s\n' % l for l in self.training_metrics)
+
+    def write_training_metric(self, delta_update_policy,
+                              delta_train_start):
+        """
+        Write training  metrics to be eventually be persisted in a  CSV
+        :param delta_update_policy: Time (in seconds) to update policy
+        :param delta_train_start: Time (in  seconds) since training started.
+        :return:
+        """
+        logger.debug("Time to update Policy: {:0.3f} s "
+                    "Time elapsed since training: {:0.3f} s "
+                    "Time for experience collection: {:0.3f} s "
+                    "Buffer Length: {} "
+                    "Returns : {:0.3f}"
```

Comment:
In my experience running early experiments, the returns shown here can be misleading when compared to rewards -- for example they sometimes start very high and then lower significantly over time whereas rewards are steadily increasing.  We might want to also include rewards here.

Reply:
Can you be a bit more specific in terms of which rewards you mean here? Do you mean cumulative rewards?